### PR TITLE
[Datasets] Update docs for drop_columns and fix typos

### LINKED ISSUE #26113: [Datasets] Add drop_column() to Dataset
We have `.add_column()` in Dataset and DatasetPipeline. In a user UX test for Datasets GA, users can easily figure out how to add a column by checking the API reference, but had difficulty to drop a column; in particular, users got surprised that there was no such API to drop a column at all.
This is a feedback we should take, because: 1) it's logical for users to expect `.drop_column()`, since there is `.add_column()`; 2) both are quite common APIs in data libraries, such as Pandas.

### PR DESCRIPTION
**Pull request by c21**
## Why are these changes needed?

We added `drop_columns()` API to datasets in https://github.com/ray-project/ray/pull/26200, so updating documentation here to use the new API - `doc/source/data/examples/nyc_taxi_basic_processing.ipynb`. In addition, fixing some minor typos after proofreading the datasets documentation.

## Related issue number

Closes https://github.com/ray-project/ray/issues/26113

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/performance-tips.rst [hunk 1]
```diff
@@ -99,7 +99,7 @@ just two of the five columns of Iris dataset.
 Parquet Row Pruning
 ~~~~~~~~~~~~~~~~~~~
 
-Similarly, you can pass in a filter to ``ray.data.read_parquet()`` (selection pushdown)
+Similarly, you can pass in a filter to ``ray.data.read_parquet()`` (filter pushdown)
```
Comment (c21) [inline doc/source/data/performance-tips.rst:102]: selection pushdown is confusing in data world, as it normally means projection to me. Other systems (such as Spark, Presto, Parquet, etc) are always using filter pushdown. We are also using filter pushdown in other places e.g. [here](https://docs.ray.io/en/master/data/creating-datasets.html#dataset-deferred-reading)

**Review by jianoaix — APPROVED** [review]
nit: there seems a typo in API comment (closing bracket): https://sourcegraph.com/github.com/ray-project/ray@master/-/blob/python/ray/data/dataset.py?L564

**Comment by c21** [comment]
> nit: there seems a typo in API comment (closing bracket): https://sourcegraph.com/github.com/ray-project/ray@master/-/blob/python/ray/data/dataset.py?L564

@jianoaix - ah good catch, fixed it.

**Review by jianoaix — APPROVED** [review]

**Comment by jianoaix** [comment]
@clarkzinzow review or merge?

File: doc/source/data/creating-datasets.rst [hunk 2]
```diff
@@ -390,7 +390,7 @@ Supported File Formats
   treating the outermost ndarray dimension as the row dimension. See our
   :ref:`tensor data guide <datasets_tensor_support>` for more information on working
   with tensors in Datasets. Although this simple example demonstrates reading a single
-  file, note that Datasets can also read directories of JSON files, with one tensor
+  file, note that Datasets can also read directories of NumPy files, with one tensor
```
Comment (clarkzinzow) [inline doc/source/data/creating-datasets.rst:393]: Nice catch!

File: doc/source/data/performance-tips.rst (hunk 1 above)
Comment (clarkzinzow) [reply doc/source/data/performance-tips.rst:102]: Agreed here!

**Review by clarkzinzow — APPROVED** [review]
LGTM, and many thanks for the drivebys!

**Comment by c21** [comment]
Thank you @clarkzinzow and @jianoaix for review!

**Pull request merged**